### PR TITLE
API cleanup

### PR DESCRIPTION
**Pull request by depl0y**
API Cleanup, extracted methods from the API module to separate files
Updated the references to API calls in other files to reflect the new changes

### CODE REVIEW

File: lib/discordrb/api/user.rb [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+      __method__,
+      :get,
+      "#{Discordrb::API.api_base}/users/#{user_id}",
+      Authorization: token
+    )
+  end
+
+  # Get profile data
+  # https://discordapp.com/developers/docs/resources/user#get-current-user
+  def resolve_current(token)
```
Comment (meew0) [inline lib/discordrb/api/user.rb]: Can you rename this method back to `profile`? It doesn't do any actual resolving, so the name is misleading here.

File: lib/discordrb/api/user.rb [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+      "#{Discordrb::API.api_base}/guilds/#{server_id}/members/@me/nick",
+      { nick: nick }.to_json,
+      Authorization: token,
+      content_type: :json
+    )
+  end
+
+  # Update user data
+  # https://discordapp.com/developers/docs/resources/user#modify-current-user
+  def update_current(token, email, password, new_username, avatar, new_password = nil)
```
Comment (meew0) [inline lib/discordrb/api/user.rb]: For consistency with my comment above, this should be `update_profile`.

File: lib/discordrb/api/server.rb [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+      "#{Discordrb::API.api_base}/guilds/#{server_id}/channels",
+      { id: channel_id, position: position }.to_json,
+      Authorization: token,
+      content_type: :json
+    )
+  end
+
+  # Get a member's data
+  # https://discordapp.com/developers/docs/resources/guild#get-guild-member
+  def member(token, server_id, user_id)
```
Comment (meew0) [inline lib/discordrb/api/server.rb]: This method, on the other hand, *does* do resolving, so it should be `resolve_member`.

File: lib/discordrb/api/server.rb [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+      __method__,
+      :get,
+      "#{Discordrb::API.api_base}/guilds/#{server_id}/members?limit=#{limit}#{"&after=#{after}" if after}",
+      Authorization: token
+    )
+  end
+
+  # Update a user properties
+  # https://discordapp.com/developers/docs/resources/guild#modify-guild-member
+  def update_user(token, server_id, user_id, nick: nil, roles: nil, mute: nil, deaf: nil, channel_id: nil)
```
Comment (meew0) [inline lib/discordrb/api/server.rb]: This should be `update_member` for consistency with the methods above.

File: lib/discordrb/api/server.rb [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+        channel_id: channel_id
+      }.reject { |_, v| v.nil? }.to_json,
+      Authorization: token,
+      content_type: :json
+    )
+  end
+
+  # Remove user from server
+  # https://discordapp.com/developers/docs/resources/guild#remove-guild-member
+  def remove_user(token, server_id, user_id)
```
Comment (meew0) [inline lib/discordrb/api/server.rb]: `remove_member`, see above

File: lib/discordrb/cache.rb [hunk 6]
```diff
@@ -59,7 +62,7 @@ def user(id)
 
       LOGGER.out("Resolving user #{id}")
       begin
-        response = API.user(token, id)
+        response = API::API.resolve(token, id)
```
Comment (meew0) [inline lib/discordrb/cache.rb]: This looks wrong, shouldn't it be `API::User`?

Comment (depl0y) [inline lib/discordrb/cache.rb]: You are right, should probably have tests ready for this.

**Comment by Daniel-Worrall** [comment]
This merge has caused a problem with `Profile#avatar=` as it calls `update_profile_data` which calls `API#update_user` which was removed in this merge.

```
E:/Ruby22-x64/lib/ruby/gems/2.2.0/gems/discordrb-2.1.3/lib/discordrb/data.rb:663:in `update_profile_data': undefined method `update_user' for Discordrb::API:Module (NoMethodError)
    from E:/Ruby22-x64/lib/ruby/gems/2.2.0/gems/discordrb-2.1.3/lib/discordrb/data.rb:641:in `avatar='
    from E:/Documents/Ruby Projects/HS/Summon Stone/stone.rb:149:in `block (2 levels) in search'
    from E:/Documents/Ruby Projects/HS/Summon Stone/stone.rb:146:in `each'
    from E:/Documents/Ruby Projects/HS/Summon Stone/stone.rb:146:in `block in search'
    from E:/Documents/Ruby Projects/HS/Summon Stone/stone.rb:139:in `each'
    from E:/Documents/Ruby Projects/HS/Summon Stone/stone.rb:139:in `search'
    from E:/Documents/Ruby Projects/HS/Summon Stone/stone.rb:179:in `process_summon'
    from E:/Documents/Ruby Projects/HS/Summon Stone/stone.rb:192:in `process_event'
    from E:/Documents/Ruby Projects/HS/Summon Stone/stone.rb:200:in `block in <top (required)>'
    from E:/Documents/Ruby Projects/HS/Summon Stone/stone.rb:198:in `loop'
    from E:/Documents/Ruby Projects/HS/Summon Stone/stone.rb:198:in `<top (required)>'
    from -e:1:in `load'
    from -e:1:in `<main>'
```

**Pull request merged**